kitchen/src/kitchen/core.clj

```diff
@@ -394,22 +403,24 @@
   [{:keys [uri] :as request}]
   (try
     (swap! total-http-requests inc)
-    (let [response (case uri
-                     "/async/request" (async-request-handler request)
-                     "/async/result" (async-result-handler request)
-                     "/async/status" (async-status-handler request)
-                     "/chunked" (chunked-handler request)
-                     "/die" (die-handler request)
-                     "/environment" (environment-handler request)
-                     "/gzip" (gzip-handler request)
-                     "/kitchen-state" (state-handler request)
-                     "/pi" (pi-handler request)
-                     "/request-info" (request-info-handler request)
-                     "/status-400" (bad-status-handler 400)
-                     "/status-401" (bad-status-handler 401)
-                     "/status-402" (bad-status-handler 402)
-                     "/unchunked" (unchunked-handler request)
-                     (default-handler request))]
+    (let [response (cond
+                     (= "/async/request" uri) (async-request-handler request)
+                     (= "/async/result" uri) (async-result-handler request)
+                     (= "/async/status" uri) (async-status-handler request)
+                     (= "/chunked" uri) (chunked-handler request)
+                     (= "/die" uri) (die-handler request)
+                     (= "/environment" uri) (environment-handler request)
+                     (= "/gzip" uri) (gzip-handler request)
+                     (= "/kitchen-state" uri) (state-handler request)
+                     (= "/pi" uri) (pi-handler request)
+                     (= "/request-info" uri) (request-info-handler request)
+                     (= "/unchunked" uri) (unchunked-handler request)
+                     (str/starts-with? uri "/sleep-") (let [uri-params (map #(Integer/parseInt %) (rest (str/split uri #"-")))]
```

Comment:
The parameter should be passed as part of the query string, not encoding in the path.